components/s3/tests/s3-component.spec.ts

```diff
@@ -216,30 +255,246 @@ describe('when downloading objects', () => {
       sendMock.mockRejectedValueOnce(new Error('S3 connection error'))
     })
 
+    afterEach(() => {
+      jest.resetAllMocks()
+    })
+
     it('should throw error with failure message', async () => {
-      await expect(component.downloadObject(key)).rejects.toThrow('S3 connection error')
+      await expect(component.downloadObjectAsString(key)).rejects.toThrow('S3 connection error')
     })
 
     it('should send GetObjectCommand with correct bucket and key before failing', async () => {
-      await expect(component.downloadObject(key)).rejects.toThrow()
+      await expect(component.downloadObjectAsString(key)).rejects.toThrow()
 
       expect(sendMock).toHaveBeenCalledTimes(1)
-      expect(sendMock).toHaveBeenCalledWith(new DeleteObjectCommand({
+      expect(sendMock).toHaveBeenCalledWith(new GetObjectCommand({
```

Comment:
I haven't noticed before, but if we're asserting the whole `getObjectCommand` object, do we need to mock it in the first place? We could use the real one, right?